### PR TITLE
Add Agentic RAG

### PR DESCRIPTION
**Pull request by aymeric-roucher**
# What does this PR do?

Add a notebook on agentic RAG.

### CODE REVIEW

File: notebooks/en/agent_rag.ipynb [hunk 1]
```diff
@@ -0,0 +1,2800 @@
+{
```
Comment (merveenoyan) [inline notebooks/en/agent_rag.ipynb:1]: <p>You switch between I and we often, try to be consistent</p><p>also dependencies*</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/135/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (merveenoyan) [inline notebooks/en/agent_rag.ipynb:1]: <p>maybe explain why you picked thenlper/gte-small?</p><p>the rest is self explanatory for someone who read your previous cookbook I guess</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/135/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (merveenoyan) [inline notebooks/en/agent_rag.ipynb:1]: <p>why not use Llama 3? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/135/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Review by merveenoyan — APPROVED** [review]
TIL agents + RAG, thanks for the recipe!
Only left minor comments

File: notebooks/en/agent_rag.ipynb (hunk 1 above)
Comment (aymeric-roucher) [reply notebooks/en/agent_rag.ipynb:1]: Thank you, I changed the phrasing to only use "we"!

Comment (aymeric-roucher) [reply notebooks/en/agent_rag.ipynb:1]: Thank you, I added a more complete evaluation part with the comparison of agentic vs standard RAG (scores goes from 70% to 78.5%! 🚀)

**Pull request merged**